webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/RudderUserDetailsFile.scala

```diff
@@ -554,8 +554,13 @@ object UserFileProcessing {
               // accept both "role" and "roles"
               userRoles(node.attribute("role")) ++ userRoles(node.attribute("roles"))
             ) match {
-              case (Some(name :: Nil), Some(pwd :: Nil), roles) if (name.size > 0) =>
-                Some(ParsedUser(name, pwd, roles)).succeed
+              case (Some(name :: Nil), pwd, roles) if (name.size > 0) =>
+                // password can be optional when an other authentication backend is used.
+                // If the attribute is defined several time, use the first occurrence.
+                pwd match {
+                  case Some(p :: _) => Some(ParsedUser(name, p, roles)).succeed
+                  case _            => Random.nextString(20).map(s => Some(ParsedUser(name, s, roles)))
```

Comment:
Random is too predictable, we should do something like https://github.com/Normation/rudder/blob/2aba894711bda3be34660005363eaf1ff316ea15/webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/api/TokenGenerator.scala#L51

Reply:
it's not java random, but let me check what actual implementation it is using

Reply:
that's already what we used for API tokens before, it uses Java Random https://github.com/scala/scala/blob/v2.12.4/src/library/scala/util/Random.scala#L29

Reply:
but in doubt and for homogenity, let's go for TokenGeneratorImpl

Reply:
(well, its logic at leat)

Reply:
:+1:

Reply:
Can we log that this user is disabled somehow ?

Reply:
Maybe we should not log about that